pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_TNCuda_Param.cpp

```diff
@@ -0,0 +1,1658 @@
+    expected_results[1][2] = {0.34958337, -0.05283436};
+    expected_results[1][5] = {0.34958337, -0.05283436};
+    expected_results[1][7] = {0.34958337, -0.05283436};
+
+    for (auto &vec : expected_results) {
+        for (auto &val : vec) {
+            val = inverse ? std::conj(val) : val;
+        }
+    }
+
+    SECTION("Apply adjacent wires") {
+
+        tn_state->reset();
+
+        tn_state->applyOperations({"Hadamard", "Hadamard", "Hadamard"},
+                                  {{0}, {1}, {2}}, {false, false, false});
+
+        tn_state->applyOperation("IsingZZ", {0, 1}, inverse, angles);
+
+        if constexpr (std::is_same_v<TNDevice_T, MPSTNCuda<double>> ||
+                      std::is_same_v<TNDevice_T, MPSTNCuda<float>>) {
+            tn_state->append_mps_final_state();
+        }
+
+        auto results = tn_state->getDataVector();
+
+        CHECK(results == Pennylane::Util::approx(expected_results[0]));
+    }
+
+    SECTION("Apply non-adjacent wires") {
+
+        tn_state->applyOperations({"Hadamard", "Hadamard", "Hadamard"},
+                                  {{0}, {1}, {2}}, {false, false, false});
+
+        tn_state->applyOperation("IsingZZ", {0, 2}, inverse, angles);
+
+        if constexpr (std::is_same_v<TNDevice_T, MPSTNCuda<double>> ||
+                      std::is_same_v<TNDevice_T, MPSTNCuda<float>>) {
+            tn_state->append_mps_final_state();
+        }
+
+        auto results = tn_state->getDataVector();
+
+        CHECK(results == Pennylane::Util::approx(expected_results[1]));
+    }
+}
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::Gates::CRX", "[TNCuda_Param]",
+                        TestTNBackends) {
+    const bool inverse = GENERATE(false, true);
+    using TNDevice_T = TestType;
+    using cp_t = typename TNDevice_T::ComplexT;
+    using Precision_T = typename TNDevice_T::PrecisionT;
+    std::size_t num_qubits = 3;
+    std::size_t maxExtent = 2;
+    DevTag<int> dev_tag{0, 0};
+
+    std::unique_ptr<TNDevice_T> tn_state =
+        createTNState<TNDevice_T>(num_qubits, maxExtent, dev_tag);
+
+    const std::vector<Precision_T> angles = {0.3};
+
+    // Results collected from `default.qubit`
+    std::vector<std::vector<cp_t>> expected_results{
+        std::vector<cp_t>(1 << num_qubits, {0.35355339, 0.0}),
+        std::vector<cp_t>(1 << num_qubits, {0.35355339, 0.0}),
+    };
+
+    expected_results[0][4] = {0.34958337, -0.05283436};
+    expected_results[0][5] = {0.34958337, -0.05283436};
+    expected_results[0][6] = {0.34958337, -0.05283436};
+    expected_results[0][7] = {0.34958337, -0.05283436};
+
+    expected_results[1][4] = {0.34958337, -0.05283436};
+    expected_results[1][5] = {0.34958337, -0.05283436};
+    expected_results[1][6] = {0.34958337, -0.05283436};
+    expected_results[1][7] = {0.34958337, -0.05283436};
+
+    for (auto &vec : expected_results) {
+        for (auto &val : vec) {
+            val = inverse ? std::conj(val) : val;
+        }
+    }
```

Comment:
If my old eyes are not tricking me, the two vectors are the same, right?
In this case, can we have it defined a single time and copied after that?

Reply:
I changed the test because if it produces the same values changing the condition of the test then the test design is incorrect due to not reflecting the different conditions of the test.